src/ilcdlib/mapping/impacts.py

```diff
@@ -48,31 +48,42 @@ class ImpactsKeywordToOpenIdMapper(KeyValueMapper[str]):
     KV = {
         "gwp-biogenic": ["biogenic"],
         "gwp-luluc": ["luluc"],
-        "gwp-fossil": ["fossil"],
-        "gwp-gwp_nonCO2": ["CO2"],
-        "gwp": ["gwp", "global warming"],
+        "gwp-nonCO2": ["CO2"],
         "ep-marine": ["marine"],
-        "ep-fresh": ["fresh", "fw"],
-        "ep-terr": ["terr"],
-        "ep": ["ep", "eutrophication"],
+        "ep-fresh": ["freshwater", "fw"],
+        "ep-terr": ["terrestrial"],
         "odp": ["odp", "depletion"],
         "ap": ["ap", "acidification"],
         "pocp": ["pocp", "photochemical", "smog", "ozone creation"],
     }
 
 
+class ImpactsRegexToOpenIdMapper(RegexMapper[str]):
+    """Map impact names using regex.""" ""
+
+    PATTERNS = {
+        "gwp-fossil": r"^(?!.*\bnon\b)(?=.*\b(fossil)\b)(?=.*\b(gwp|global warming)\b).*$",
+        "gwp": r"^(?!.*\b(fossil|luluc|CO2|total)\b)(?=.*\b(gwp|global warming)\b).*$",
+        "ep": r"^(?!.*\b(freshwater|fw|marine|terr)\b)(?=.*\b(ep|eutrophication)\b).*$",
```

Comment:
There is no such thing as just `ep`. It should be one of 
![image](https://github.com/cchangelabs/ilcdlib/assets/5593615/9ce18c4b-3939-4a9d-892b-bf3ed0463e8e)